source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+.. code-block:: bash
+
+   INFO:    Could not find any nv files on this host!
+   hello world
+
+as the UI nodes do not have access to GPUs and thus do not have an nv file to point the container to the required libraries. Running the script in batch mode with ``sbatch script.sh``, the ``-p`` flag is used, and the job ends up on a GPU node. The output becomes:
+
+.. code-block:: bash
+
+   hello world
+
+Of course, this ubuntu image does not have any of the tools needed to build GPU-native code or libraries that can run on the GPU. Refer to :ref:`this section <resources-singularity>` for more resources.
+
+.. tip::
+
+   While you do not get the warning about finding the nv file when using the ``--nv`` flag, you also have to specify which GPU to use, otherwise none are allocated to you! This can be done with the ``-G`` flag, as can be seen in the example shell script. 
+
+Now you are ready to build on top of a base container and run your code on a GPU!
+
+.. _accounting-gpu:
+
+Accounting of GPU usage
+=======================
+
+Currently the usage of GPU nodes is accounted for in GPU hours. This means that even though multiple cores are used simultaneously, one hour of use of a GPU node is billed as 1 GPU-hour. Unlike CPUs where one hour of multi-core usage is billed as multiple CPU hours, depending on the number of cores.
+
+.. _singularity-building:
+
+================================
+Building and running a singularity container
+================================
+
+In this section we show how to build a singularity container use it to run code in its environment. There is extensive documentation from singularity itself `here <https://docs.sylabs.io/guides/latest/user-guide/index.html>`_. 
+
+The steps in this section are done on GPU nodes, to ensure availability of the drivers, which may be needed in some compilation steps.
+
+Building directly from dockerhub
+================================
+
+There are multiple ways to build a container. To build directly from docker hub, for example the latest version of tensorflow, one can invoke:
+
+.. code-block:: bash
+   
+   singularity build --nv tf_latest.sif docker://tensorflow/tensorflow:latest
+
+and the image ``tf_latest.sif`` from `dockerhub <https://hub.docker.com>`_ will be built, containing the contents of the latest ``tensorflow`` image from the makers of tensorflow. You can also get an image from a different source, such as the Nvidia container repository:
+
+.. code-block:: bash
+
+   singularity build --nv nvidia-tf.sif docker://nvcr.io/nvidia/tensorflow:22.07-tf2-py3
+
+Such an image will contain all the necessary drivers and compile flags to run on Nvidia GPUs, which is preferable on :abbr:`Spider (Symbiotic Platform(s) for Interoperable Data Extraction and Redistribution)`.
```

Comment:
What do you mean by 'Such an image'. if you mean that an image from the nvidia repo is pre-built to have all the necessary drivers etc then perhaps clarify this (and are you are that this is true for 'all' nvidia pre-build containers - if this is described by nvidia then perhaps link it). For the confusion here comes from the fact that there is no clear distinction in the text between the tensorflow example and the nvidia example. Hence if the nvidia repo has some benefits that we suggest users need to be aware then make this very clear.